lib/parse.js

```diff
@@ -335,7 +335,11 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
         if (prefix) num = prefix + num;
         var valid = parse_js_number(num);
         if (!isNaN(valid)) {
-            return token("num", valid);
+            var tok = token("num", valid);
+            if (num.indexOf('.') >= 0) {
+                tok.literal = num;
```

Comment:
What do you think about this - if `tok.literal = num;` is always set for every number (integer or float) and if there is no scope for a given AST_Number instance, then the original numeric string literal could be output unconditionally.  Then the asm.js blocks would still generate correct code even without scope.  The downside to this is that minified code without compression or mangling would not have compacted numbers as they presently do.